app/WebParsing/PostParser.hs

```diff
@@ -20,46 +20,23 @@ fasCalendarURL = "http://calendar.artsci.utoronto.ca/"
 failedString :: String
 failedString = "Failed."
 
-getPost :: T.Text -> IO ()
-getPost str = do
-    let path = fasCalendarURL ++ (T.unpack str)
-    rsp <- simpleHTTP (getRequest path)
-    body <- getResponseBody rsp
-    let tags = filter isNotComment $ parseTags body
-        postsSoup = secondH2 tags
-        posts = partitions isPostName postsSoup
-    runSqlite databasePath $ do
-        runMigration migrateAll
-        mapM_ addPostToDatabase posts
-    print $ "parsing " ++ (T.unpack str)
-    where
-        isNotComment (TagComment _) = False
-        isNotComment _ = True
-        secondH2 tags =
-            let sect = sections (isTagOpenName "h2") tags
-            in
-                if (length sect) < 2
-                then
-                    []
-                else
-                    takeWhile isNotCoursesSection tags
-        isNotCoursesSection tag = not (tagOpenAttrLit "a" ("name", "courses") tag)
-        isPostName tag = tagOpenAttrNameLit "a" "name" (\nameValue -> (length nameValue) == 9) tag
-
-addPostToDatabase :: [Tag String] -> SqlPersistM ()
-addPostToDatabase tags = do
-    let postCode_ = T.pack (fromAttrib "name" ((take 1 $ filter (isTagOpenName "a") tags) !! 0))
-        liPartitions = partitions isLiTag tags
-        programPrereqs = map getCourseFromTag $ map (T.pack . fromAttrib "href") $ filter isCourseTag tags
+addPostToDatabase :: [Tag T.Text] -> SqlPersistM ()
+addPostToDatabase programElements = do
+    -- TODO: Remove Focuses from programElements
+    let fullPostName = innerText $ take 1 $ filter isTagText programElements
```

Comment:
`innerText` needs a list of Tags, which is what `take 1` gives you, but `head` only gives the element itself.